modules/core/src/main/java/org/apache/synapse/config/xml/ScatterGatherMediatorFactory.java

```diff
@@ -0,0 +1,185 @@
+import org.apache.synapse.SynapseException;
+import org.apache.synapse.mediators.eip.Target;
+import org.apache.synapse.mediators.v2.ScatterGather;
+import org.jaxen.JaxenException;
+
+import java.util.Iterator;
+import java.util.Properties;
+import javax.xml.namespace.QName;
+
+/**
+ * The &lt;scatter-gather&gt; mediator is used to copy messages in Synapse to similar messages but with
+ * different message contexts and aggregate the responses back.
+ *
+ * <pre>
+ * &lt;scatter-gather parallel-execution=(true | false) result-target=(body | variable) content-type=(JSON | XML)&gt;
+ *   &lt;aggregation value="expression" condition="expression" timeout="long"
+ *     min-messages="expression" max-messages="expression"/&gt;
+ *   &lt;sequence&gt;
+ *     (mediator)+
+ *   &lt;/sequence&gt;+
+ * &lt;/scatter-gather&gt;
+ * </pre>
+ */
+public class ScatterGatherMediatorFactory extends AbstractMediatorFactory {
+
+    /**
+     * This will hold the QName of the clone mediator element in the xml configuration
+     */
+    private static final QName SCATTER_GATHER_Q
+            = new QName(XMLConfigConstants.SYNAPSE_NAMESPACE, "scatter-gather");
+    private static final QName ELEMENT_AGGREGATE_Q
+            = new QName(XMLConfigConstants.SYNAPSE_NAMESPACE, "aggregation");
+    private static final QName ATT_AGGREGATE_EXPRESSION = new QName("expression");
+    private static final QName ATT_CONDITION = new QName("condition");
+    private static final QName ATT_TIMEOUT = new QName("timeout");
+    private static final QName ATT_MIN_MESSAGES = new QName("min-messages");
+    private static final QName ATT_MAX_MESSAGES = new QName("max-messages");
+    private static final QName SEQUENCE_Q = new QName(XMLConfigConstants.SYNAPSE_NAMESPACE, "sequence");
+    private static final QName PARALLEL_EXEC_Q = new QName("parallel-execution");
+    private static final QName RESULT_TARGET_Q = new QName("result-target");
+    private static final QName ROOT_ELEMENT_Q = new QName("root-element");
+    private static final QName CONTENT_TYPE_Q = new QName("content-type");
+
+    private static final SequenceMediatorFactory fac = new SequenceMediatorFactory();
+
+    public Mediator createSpecificMediator(OMElement elem, Properties properties) {
+
+        boolean asynchronousExe = true;
+
+        ScatterGather mediator = new ScatterGather();
+        processAuditStatus(mediator, elem);
+
+        OMAttribute parallelExecAttr = elem.getAttribute(PARALLEL_EXEC_Q);
+        if (parallelExecAttr != null && parallelExecAttr.getAttributeValue().equals("false")) {
+            asynchronousExe = false;
+        }
+        mediator.setParallelExecution(asynchronousExe);
+
+        OMAttribute contentTypeAttr = elem.getAttribute(CONTENT_TYPE_Q);
+        if (contentTypeAttr == null || StringUtils.isBlank(contentTypeAttr.getAttributeValue())) {
+            String msg = "The 'content-type' attribute is required for the configuration of a Scatter Gather mediator";
```

Comment:
We discussed to derive this from the message context, and make this optional right? Was that changed?

Reply:
It was discussed for the Foreach mediator when we need to update the original collection.